### PR TITLE
uni-algo 1.2.0

### PR DESCRIPTION
**Pull request by netheril96**
Add a new formula

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by netheril96** [comment]
How do I debug the CI failure? The error message contains no useful information, and I cannot reproduce it on my Linux or Mac.

**Review by chenrui333 — APPROVED** [review]

File: Formula/u/uni-algo.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  desc "Unicode Algorithms Implementation for C/C++"
+  homepage "https://github.com/uni-algo/uni-algo"
+  url "https://github.com/uni-algo/uni-algo/archive/refs/tags/v1.2.0.tar.gz"
+  sha256 "f2a1539cd8635bc6088d05144a73ecfe7b4d74ee0361fabed6f87f9f19e74ca9"
+  license "MIT"
+
+  depends_on "cmake" => [:build, :test]
+
+  def install
+    system "cmake", "-DBUILD_SHARED_LIBS=ON", "-S", ".", "-B", "build", *std_cmake_args
```
Comment (chenrui333) [inline Formula/u/uni-algo.rb:11]:
```suggestion
    system "cmake", "-S", ".", "-B", "build", "-DBUILD_SHARED_LIBS=ON", *std_cmake_args
```

File: Formula/u/uni-algo.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    (testpath/"utf8_norm.cpp").write <<~EOS
+      #include <uni_algo/norm.h>
+      int main() {
+        return (una::norm::to_nfc_utf8("W\u0302") == "Ŵ") ? 0 : 1;
+      }
+    EOS
+
+    system "cmake", ".", "-DCMAKE_PREFIX_PATH:STRING=#{opt_lib}", "-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON"
+    system "make"
+    system "./utf8_norm"
```
Comment (chenrui333) [inline Formula/u/uni-algo.rb:34]: maybe assert the response?

File: Formula/u/uni-algo.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    (testpath/"utf8_norm.cpp").write <<~EOS
+      #include <uni_algo/norm.h>
+      int main() {
+        return (una::norm::to_nfc_utf8("W\u0302") == "Ŵ") ? 0 : 1;
+      }
+    EOS
+
+    system "cmake", ".", "-DCMAKE_PREFIX_PATH:STRING=#{opt_lib}", "-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON"
+    system "make"
```
Comment (chenrui333) [inline Formula/u/uni-algo.rb:33]:
```suggestion
    system "cmake", "-S", ".", "-B", "build", "-DCMAKE_PREFIX_PATH:STRING=#{opt_lib}", "-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON"
    system "cmake", "--build", "build"
```

**Comment by chenrui333** [comment]
overall looks good, couple of comments for the formula improvements.

**Comment by chenrui333** [comment]
can you also add uni-algo into .github/autobump.txt file? Thanks!

File: Formula/u/uni-algo.rb (hunk 2 above)
Comment (netheril96) [reply Formula/u/uni-algo.rb:34]: This implicitly asserts that the exit code is 0

**Comment by netheril96** [comment]
autobump added

**Review by carlocab — APPROVED** [review]

**Review by nandahkrishna — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@netheril96, thanks for your contribution to Homebrew! :tada: 🥇

**Pull request merged**